apps/interop-rxjs-signal/src/app/list/photos.component.ts

```diff
@@ -80,40 +91,88 @@ import { PhotoStore } from './photos.store';
       </section>
     </ng-container>
   `,
-  providers: [provideComponentStore(PhotoStore)],
   host: {
     class: 'p-5 block',
   },
 })
-export default class PhotosComponent implements OnInit {
-  store = inject(PhotoStore);
-  readonly vm$ = this.store.vm$.pipe(
-    tap(({ search }) => {
-      if (!this.formInit) {
-        this.search.setValue(search);
-        this.formInit = true;
-      }
-    })
+export default class PhotosComponent extends SignalState<PhotoComponentState> {
+  private readonly photoService = inject(PhotoService);
+  protected readonly vm = this.selectMany(
+    ['photos', 'search', 'page', 'pages', 'loading', 'error'],
+    ({ photos, search, page, pages, loading, error }) => {
+      return {
+        photos,
+        search,
+        pages,
+        page,
+        loading,
+        error,
+        endOfPage: page === pages,
+      };
+    }
   );
+  protected readonly searchCtrl = new FormControl();
 
-  private formInit = false;
-  search = new FormControl();
-
-  ngOnInit(): void {
-    this.store.search(
-      this.search.valueChanges.pipe(
-        skipWhile(() => !this.formInit),
-        debounceTime(300),
-        distinctUntilChanged()
+  constructor() {
+    super();
+    this.initialize({
+      photosResult: undefined,
+      photos: [],
+      search: '',
+      page: 1,
+      pages: 1,
+      loading: false,
+      error: '',
+    });
+    const photosResult$ = toObservable(
+      this.selectMany(['search', 'page'])
+    ).pipe(
+      debounceTime(300),
+      tap(() => this.patch({ loading: true })),
+      switchMap(({ search, page }) =>
+        this.photoService
+          .searchPublicPhotos(search, page)
+          .pipe(finalize(() => this.patch({ loading: false })))
       )
     );
+    const photos = this.select(
+      'photosResult',
+      (photosResult) => photosResult?.photos.photo as Photo[]
+    );
+
+    this.connect({
+      search: toSignal(this.searchCtrl.valueChanges, {
+        initialValue: this.searchCtrl.value,
+      }),
+      photosResult: toSignal(photosResult$),
+      photos,
+    });
+
+    // I would prefer to call a search function for the template
+    // but it is to prove the concept of effects in the signalstate
+    effect(
+      () => {
+        if (this.select('search')) {
+          this.patch({ page: 1 });
+        }
+      },
+      { allowSignalWrites: true }
+    );
+  }
```

Comment:
I don't get what you are trying to prove ? 😅
But indeed a search function will be way easier to understand, and it will not need to write `allowSignalWrite` which indicate a bad pattern, IMO.

Reply:
Agreed, It's needed here because it is an effect, but a function called in the template would be clearer and more declarative imo